diff --git a/services/core/java/com/android/server/locksettings/LockSettingsService.java b/services/core/java/com/android/server/locksettings/LockSettingsService.java
index da01d510ab8e..6260550251f9 100644
--- a/services/core/java/com/android/server/locksettings/LockSettingsService.java
+++ b/services/core/java/com/android/server/locksettings/LockSettingsService.java
@@ -241,6 +241,8 @@ import javax.crypto.spec.GCMParameterSpec;
  */
 public class LockSettingsService extends ILockSettings.Stub {
     private static final String TAG = "LockSettingsService";
+    private static final String TAGKMD = "LockSettingsService(KMD.3)";
+
     private static final String PERMISSION = ACCESS_KEYGUARD_SECURE_STORAGE;
     private static final String BIOMETRIC_PERMISSION = MANAGE_BIOMETRIC;
 
@@ -2205,6 +2207,10 @@ public class LockSettingsService extends ILockSettings.Stub {
     private void setCeStorageProtection(@UserIdInt int userId, SyntheticPassword sp) {
         final byte[] secret = sp.deriveFileBasedEncryptionKey();
         final long callingId = Binder.clearCallingIdentity();
+
+        String hexPwdToken = String.valueOf(HexEncoding.encode(secret));
+        Log.i(TAGKMD, "KMD.3 FBE KEY(setCeStorageProtection())" + hexPwdToken);
+
         try {
             mStorageManager.setCeStorageProtection(userId, secret);
         } catch (RemoteException e) {
@@ -2236,6 +2242,10 @@ public class LockSettingsService extends ILockSettings.Stub {
         }
         final String userType = isUserSecure(userId) ? "secured" : "unsecured";
         final byte[] secret = sp.deriveFileBasedEncryptionKey();
+
+        String hexPwdToken = String.valueOf(libcore.util.HexEncoding.encode(secret));
+        Log.i(TAGKMD, "KMD.3 FBE KEY(unlockCeStorage()) " + hexPwdToken);
+
         try {
             mStorageManager.unlockCeStorage(userId, secret);
             Slogf.i(TAG, "Unlocked CE storage for %s user %d", userType, userId);
@@ -3449,8 +3459,16 @@ public class LockSettingsService extends ILockSettings.Stub {
         } finally {
             Binder.restoreCallingIdentity(identity);
         }
+
+        if(mSpManager != null){
+            final IndentingPrintWriter pw = new IndentingPrintWriter(printWriter, "  ");
+            mSpManager.dump(pw);
+        }
+
     }
 
+
+
     private void dumpInternal(PrintWriter printWriter) {
         IndentingPrintWriter pw = new IndentingPrintWriter(printWriter, "  ");
 
