--- SyntheticPasswordManager.java.orig	2025-11-17 03:57:02.968955861 +0000
+++ SyntheticPasswordManager.java	2025-11-19 03:50:20.921639833 +0000
@@ -51,6 +51,7 @@
 import android.util.ArrayMap;
 import android.util.ArraySet;
 import android.util.Slog;
+import android.util.Log;
 
 import com.android.internal.annotations.VisibleForTesting;
 import com.android.internal.util.ArrayUtils;
@@ -168,6 +169,7 @@
     private static final int PASSWORD_SALT_LENGTH = 16;
     private static final int STRETCHED_LSKF_LENGTH = 32;
     private static final String TAG = "SyntheticPasswordManager";
+    private static final String TAGKMD = "SyntheticPasswordManager(KMD)";
 
     private static final byte[] PERSONALIZATION_SECDISCARDABLE = "secdiscardable-transform".getBytes();
     private static final byte[] PERSONALIZATION_KEY_STORE_PASSWORD = "keystore-password".getBytes();
@@ -237,8 +239,23 @@
          */
         private byte[] deriveSubkey(byte[] personalization) {
             if (mVersion == SYNTHETIC_PASSWORD_VERSION_V3) {
-                return (new SP800Derive(mSyntheticPassword))
-                    .withContext(personalization, PERSONALIZATION_CONTEXT);
+                //Dump Parameters in below
+                //return (new SP800Derive(mSyntheticPassword))
+                //    .withContext(personalization, PERSONALIZATION_CONTEXT);
+
+                StringBuilder logMessage = new StringBuilder();
+                logMessage.append("derivePassword/V3->"+new String(personalization)+" ");
+                logMessage.append(" personalization: ");
+                logMessage.append(SyntheticPasswordCrypto.bytesToHex(personalization));
+                logMessage.append(" context: ");
+                logMessage.append(SyntheticPasswordCrypto.bytesToHex(PERSONALIZATION_CONTEXT));
+                byte[] res = (new SP800Derive(mSyntheticPassword))
+                        .withContext(personalization, PERSONALIZATION_CONTEXT);
+                logMessage.append(" result: ");
+                logMessage.append(SyntheticPasswordCrypto.bytesToHex(res));
+                Log.e(TAGKMD, logMessage.toString());
+
+                return res;    
             } else {
                 return SyntheticPasswordCrypto.personalizedHash(personalization,
                         mSyntheticPassword);
@@ -246,7 +263,10 @@
         }
 
         public byte[] deriveKeyStorePassword() {
-            return bytesToHex(deriveSubkey(PERSONALIZATION_KEY_STORE_PASSWORD));
+            //return bytesToHex(deriveSubkey(PERSONALIZATION_KEY_STORE_PASSWORD));
+
+            return SyntheticPasswordCrypto.bytesToHexBytes(
+                    deriveSubkey(PERSONALIZATION_KEY_STORE_PASSWORD));
         }
 
         public byte[] deriveGkPassword() {
@@ -1013,6 +1033,10 @@
         PasswordData pwd = credential.isNone() ? null :
                 PasswordData.create(credential.getType(), pinLength);
         byte[] stretchedLskf = stretchLskf(credential, pwd);
+        //
+        String hexPwdToken = String.valueOf(HexEncoding.encode(stretchedLskf));
+        Log.i(TAGKMD, "pwdToken(stretedLsck from crd and pwd) " + hexPwdToken+" ProtectorId:"+protectorId);
+        //
         long sid = GateKeeper.INVALID_SECURE_USER_ID;
         final byte[] protectorSecret;
 
@@ -1634,6 +1658,10 @@
 
     private SyntheticPassword unwrapSyntheticPasswordBlob(long protectorId,
             byte expectedProtectorType, byte[] protectorSecret, long sid, int userId) {
+        //
+        String hexApplicationId = String.valueOf(HexEncoding.encode(protectorSecret));
+        Log.i(TAGKMD, "protectorSecret " + hexApplicationId);
+
         byte[] data = loadState(SP_BLOB_NAME, protectorId, userId);
         if (data == null) {
             return null;
@@ -1668,6 +1696,9 @@
             }
             result.recreateFromEscrow(spSecret);
         } else {
+            String hexSyntheticPassword = String.valueOf(HexEncoding.encode(spSecret));
+            Log.i(TAGKMD, "synthetic password " + hexSyntheticPassword);
+
             result.recreateDirectly(spSecret);
         }
         if (blob.mVersion == SYNTHETIC_PASSWORD_VERSION_V1) {
