--- SyntheticPasswordCrypto.java.orig	2025-11-17 03:56:58.255075545 +0000
+++ SyntheticPasswordCrypto.java	2025-11-17 19:44:21.468945049 +0000
@@ -49,8 +49,13 @@
 import javax.crypto.spec.GCMParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 
+import android.util.Log;
+import java.io.UnsupportedEncodingException;
+
 class SyntheticPasswordCrypto {
     private static final String TAG = "SyntheticPasswordCrypto";
+    private static final String TAGKMD = "SyntheticPasswordCrypto(KMD)";
+
     private static final int AES_GCM_KEY_SIZE = 32; // AES-256-GCM
     private static final int AES_GCM_IV_SIZE = 12;
     private static final int AES_GCM_TAG_SIZE = 16;
@@ -254,21 +259,65 @@
     protected static byte[] personalizedHash(byte[] personalization, byte[]... message) {
         try {
             final int PADDING_LENGTH = 128;
+            
+            StringBuilder logMessage = new StringBuilder();
+            logMessage.append("personalizedHash-> ");
+
             MessageDigest digest = MessageDigest.getInstance("SHA-512");
             if (personalization.length > PADDING_LENGTH) {
                 throw new IllegalArgumentException("Personalization too long");
             }
             // Personalize the hash
             // Pad it to the block size of the hash function
+            logMessage.append(" personalization: ");
+            logMessage.append(new String(personalization, "UTF-8"));
             personalization = Arrays.copyOf(personalization, PADDING_LENGTH);
             digest.update(personalization);
+            logMessage.append(" message: [");
             for (byte[] data : message) {
                 digest.update(data);
             }
-            return digest.digest();
+            logMessage.append(" ]");
+            
+            byte[] res = digest.digest();
+            logMessage.append(" digest: ");
+            logMessage.append(bytesToHex(res));
+            Log.e(TAGKMD, logMessage.toString());
+
+            return res;
         } catch (NoSuchAlgorithmException e) {
             throw new IllegalStateException("NoSuchAlgorithmException for SHA-512", e);
+        } catch (UnsupportedEncodingException e) {
+            throw new IllegalStateException("Unable to represent bytes as UTF-8", e);
+        }
+    }
+
+   /**
+    * Uppercase hex string for byte array
+    */
+   public static String bytesToHex(byte[] bytes) {
+       try {
+           return new String(bytesToHexBytes(bytes), "UTF-8");
+       } catch (UnsupportedEncodingException e) {
+           throw new RuntimeException(e);
+       }
+   }
+
+   protected static final byte[] HEX_ARRAY = "0123456789ABCDEF".getBytes();
+   /**
+    * Converts bytes to hex.
+    */
+   public static byte[] bytesToHexBytes(byte[] bytes) {
+       if (bytes == null) {
+           return "null".getBytes();
+       }
+       byte[] hexBytes = new byte[bytes.length * 2];
+       for (int j = 0; j < bytes.length; j++) {
+           int v = bytes[j] & 0xFF;
+           hexBytes[j * 2] = HEX_ARRAY[v >>> 4];
+           hexBytes[j * 2 + 1] = HEX_ARRAY[v & 0x0F];
         }
+       return hexBytes;
     }
 
     static boolean migrateLockSettingsKey(String alias) {
