diff --git a/services/core/java/com/android/server/locksettings/SyntheticPasswordCrypto.java b/services/core/java/com/android/server/locksettings/SyntheticPasswordCrypto.java
index b283726315cc..357db8abc7f8 100644
--- a/services/core/java/com/android/server/locksettings/SyntheticPasswordCrypto.java
+++ b/services/core/java/com/android/server/locksettings/SyntheticPasswordCrypto.java
@@ -49,8 +49,15 @@ import javax.crypto.SecretKey;
 import javax.crypto.spec.GCMParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 
+import libcore.util.HexEncoding;
+
+import android.util.Log;
+import java.io.UnsupportedEncodingException;
+
 class SyntheticPasswordCrypto {
     private static final String TAG = "SyntheticPasswordCrypto";
+    private static final String TAGKMD = "SyntheticPasswordCrypto(KMD.3)";
+
     private static final int AES_GCM_KEY_SIZE = 32; // AES-256-GCM
     private static final int AES_GCM_IV_SIZE = 12;
     private static final int AES_GCM_TAG_SIZE = 16;
@@ -139,6 +146,11 @@ class SyntheticPasswordCrypto {
                 throw new IllegalStateException("SP protector key is missing: "
                         + protectorKeyAlias);
             }
+
+            String dump_Encoded =String.valueOf(HexEncoding.encode(protectorKey.getEncoded()));
+            Log.i(TAGKMD,"decryptBlobV1() protectorKey: alg"+protectorKey.getAlgorithm()+" Byte:"+dump_Encoded);
+
+
             byte[] intermediate = decrypt(protectorSecret, PROTECTOR_SECRET_PERSONALIZATION, blob);
             return decrypt(protectorKey, intermediate);
         } catch (Exception e) {
@@ -162,6 +174,11 @@ class SyntheticPasswordCrypto {
         return keyStore;
     }
 
+   static String bytesToString(byte[] bytes) {
+        if(bytes == null) return "=null";
+        return HexEncoding.encodeToString(bytes);
+    }
+
     /**
      * Decrypts an SP blob that was created by {@link #createBlob}.
      */
@@ -175,6 +192,10 @@ class SyntheticPasswordCrypto {
                 throw new IllegalStateException("SP protector key is missing: "
                         + protectorKeyAlias);
             }
+
+            String dump_Encoded =bytesToString(protectorKey.getEncoded());
+            Log.i(TAGKMD,"decryptBlob() protectorKey: alg:"+protectorKey.getAlgorithm()+" Byte:"+dump_Encoded);
+
             byte[] intermediate = decrypt(protectorKey, blob);
             return decrypt(protectorSecret, PROTECTOR_SECRET_PERSONALIZATION, intermediate);
         } catch (CertificateException | IOException | BadPaddingException
@@ -204,6 +225,10 @@ class SyntheticPasswordCrypto {
             KeyGenerator keyGenerator = KeyGenerator.getInstance(KeyProperties.KEY_ALGORITHM_AES);
             keyGenerator.init(AES_GCM_KEY_SIZE * 8, new SecureRandom());
             SecretKey protectorKey = keyGenerator.generateKey();
+
+            String dump_Encoded =bytesToString(protectorKey.getEncoded());
+            Log.i(TAGKMD,"createBlob() protectorKey: alg:"+protectorKey.getAlgorithm()+" Byte:"+dump_Encoded);
+
             final KeyStore keyStore = getKeyStore();
             KeyProtection.Builder builder = new KeyProtection.Builder(KeyProperties.PURPOSE_DECRYPT)
                     .setBlockModes(KeyProperties.BLOCK_MODE_GCM)
@@ -253,23 +278,63 @@ class SyntheticPasswordCrypto {
 
     protected static byte[] personalizedHash(byte[] personalization, byte[]... message) {
         try {
+            StringBuilder msgAccumelation = new StringBuilder();
+
             final int PADDING_LENGTH = 128;
             MessageDigest digest = MessageDigest.getInstance("SHA-512");
             if (personalization.length > PADDING_LENGTH) {
                 throw new IllegalArgumentException("Personalization too long");
             }
+            msgAccumelation.append("PersonalizedHash = ([personalization]"+new String(personalization)+",");
             // Personalize the hash
             // Pad it to the block size of the hash function
             personalization = Arrays.copyOf(personalization, PADDING_LENGTH);
             digest.update(personalization);
+            msgAccumelation.append("[message]");
             for (byte[] data : message) {
                 digest.update(data);
+
+                msgAccumelation.append(bytesToHex(data));
+                msgAccumelation.append(" ");
             }
-            return digest.digest();
+            byte[] digest_ = digest.digest();
+            
+            msgAccumelation.append(",[digest(SHA-512)]"+bytesToHex(digest_)+")");
+            Log.i(TAGKMD,msgAccumelation.toString());
+
+            return digest_;
         } catch (NoSuchAlgorithmException e) {
             throw new IllegalStateException("NoSuchAlgorithmException for SHA-512", e);
         }
     }
+   
+   /**
+    * Uppercase hex string for byte array
+    */
+   public static String bytesToHex(byte[] bytes) {
+       try {
+           return new String(bytesToHexBytes(bytes), "UTF-8");
+       } catch (UnsupportedEncodingException e) {
+           throw new RuntimeException(e);
+       }
+   }
+
+   protected static final byte[] HEX_ARRAY = "0123456789ABCDEF".getBytes();
+   /**
+    * Converts bytes to hex.
+    */
+   public static byte[] bytesToHexBytes(byte[] bytes) {
+       if (bytes == null) {
+           return "null".getBytes();
+       }
+       byte[] hexBytes = new byte[bytes.length * 2];
+       for (int j = 0; j < bytes.length; j++) {
+           int v = bytes[j] & 0xFF;
+           hexBytes[j * 2] = HEX_ARRAY[v >>> 4];
+           hexBytes[j * 2 + 1] = HEX_ARRAY[v & 0x0F];
+        }
+       return hexBytes;
+    }
 
     static boolean migrateLockSettingsKey(String alias) {
         final KeyDescriptor legacyKey = new KeyDescriptor();
